dmoj/problem.py

```diff
@@ -396,3 +391,25 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__.update(state)
+
+
+class BatchedTestCase(AbstractTestCase):
+    def __init__(self, position, batch, config, problem):
+        super().__init__(position, batch, config, problem)
+
+
+class StandaloneTestCase(AbstractTestCase, TopLevelCase):
+    def __init__(self, position, config, problem):
+        AbstractTestCase.__init__(self, position, None, config, problem)
+        TopLevelCase.__init__(self, config)
+
+
+class Batch(TopLevelCase):
+    def __init__(self, batch_no, config, problem, cases: List[BatchedTestCase]):
+        super().__init__(config)
+        self.batch = batch_no
```

Comment:
Whatever we call this variable, we should be consistent throughout.

Reply:
Done. I think `batch_no` is a much worse name. Let's just type the variable where it appears.